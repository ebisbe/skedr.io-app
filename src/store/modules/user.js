import Auth from '@aws-amplify/auth'
import LogRocket from 'logrocket'

export const state = {
  user: null,
  username: null,
  userId: null,
  userVerification: null
}

export const getters = {
  userId: state => state.username
}

export const mutations = {
  setUser(state, user) {
    state.user = user
    if (user !== null) {
      state.username = decodeURIComponent(user.username)
    } else {
      state.username = null
    }
  },
  setUserAttributes(state, attributes) {
    state.user.attributes = Object.assign({}, state.user.attributes, attributes)
  },
  setUserId: (state, userId) => (state.userId = userId),
  setUserVerification: (state, data) => {
    state.userVerification = data
  }
}

export const actions = {
  loginUser: async function({ state, commit }, { username, password }) {
    // the value of `username` is really an `email`
    const user = await Auth.signIn(username.toLowerCase(), password)
    commit('setUser', user)

    const data = await Auth.verifiedContact(user)
    commit('setUserVerification', data)

    const payload = {
      body: { userId: state.username, email: username }
    }
    return Object.keys(data.verified).length !== 0
  },
  currentAuthenticatedUser: async ({ commit, state }) => {
    try {
      const user = await Auth.currentAuthenticatedUser({ bypassCache: true })
      const credentials = await Auth.currentCredentials()
      commit('setUser', user)
      commit('setUserId', credentials.identityId)
      LogRocket.identify(state.username, {
        name: `${user.attributes.name} ${user.attributes.family_name}`,
        email: user.attributes.email
      })
      LogRocket.getSessionURL(sessionURL => {
        const indentifyUser = {
          name: `${user.attributes.name} ${user.attributes.family_name}`,
          email: user.attributes.email,
          sessionURL: sessionURL
        }
        Beacon('identify', indentifyUser)
      })
    } catch (err) {
      commit('setUser', null)
      return false
    }
    return true
  },
  updateUserAttributes: async ({ commit, state }) => {
    await Auth.updateUserAttributes(state.user, state.user.attributes)
    const user = await Auth.currentAuthenticatedUser({ bypassCache: true })
    commit('setUser', user)
  },
  confirmEmailCode: async ({ commit }, code) => {
    await Auth.verifyCurrentUserAttributeSubmit('email', code)
    const user = await Auth.currentAuthenticatedUser({ bypassCache: true })
    commit('setUser', user)
  },
  signOut: async ({ state }) => {
    if (state.user.attributes.email_verified === false) {
      throw {
        name: 'VerifyEmailError',
        message: 'To sign out first verify your email'
      }
    }
    await Auth.signOut()
    return true
  }
}
