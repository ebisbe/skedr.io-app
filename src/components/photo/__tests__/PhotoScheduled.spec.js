import { shallowMount } from '@vue/test-utils'
import comp from '../PhotoScheduled'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

Vue.use(Vuetify)
Vue.use(Vuex)

import { getters } from '@/store/modules/schedulerPool'
const createStore = (photos = {}) =>
  new Vuex.Store({
    modules: {
      schedulerPool: {
        namespaced: true,
        state: {
          pool: photos
        },
        getters
      }
    }
  })
const createComp = (propsData, photos = {}) =>
  shallowMount(comp, {
    propsData,
    store: createStore(photos)
  })

describe('PhotoScheduled.vue', () => {
  const photo = {
    photoId: 'photoId',
    groupId: 'groupId',
    message: 'Some message',
    photo: {
      title: 'Some title',
      secret: 'secret',
      server: 'server',
      farm: 'farm'
    }
  }

  describe('Properties', () => {
    it('requires a Photo Flickr Object', () => {
      const wrapper = createComp({ photo })
      expect(wrapper.props().photo).toBe(photo)
    })
  })

  describe('Computed Properties', () => {
    it('returns the url for a photo at flickr', () => {
      const wrapper = createComp({ photo })
      const url = 'https://farmfarm.staticflickr.com/server/photoId_secret_q.jpg'
      expect(wrapper.vm.url_q).toBe(url)
    })

    it('checks the payload', () => {
      const wrapper = createComp({ photo })
      expect(wrapper.vm.payload).toMatchObject({
        id: 'groupId-photoId',
        item: photo
      })
    })
  })

  it('matches snapshot', () => {
    const wrapper = createComp({ photo })
    expect(wrapper).toMatchSnapshot()
  })
})
