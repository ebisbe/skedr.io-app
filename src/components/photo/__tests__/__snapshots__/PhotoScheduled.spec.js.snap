// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoScheduled.vue matches snapshot 1`] = `
<base-photo-stub height="250" url="https://farmfarm.staticflickr.com/server/photoId_secret.jpg" lazyurl="https://farmfarm.staticflickr.com/server/photoId_secret_q.jpg" isprivate="true" selectable="true" class="">
  <!---->
  <footer-photo-stub title="Some title" subtitle=""></footer-photo-stub>
</base-photo-stub>
`;
